### PR TITLE
Fix KPO to have hyphen instead of period

### PR DESCRIPTION
**Pull request by dimon222**
I don't know how it got missed after all the changes done, but the main logic still wasn't applied to have no period afterall.

In other words, the main single character change wasn't done to achieve the purpose. See changes included.
https://github.com/apache/airflow/pull/19036

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
Just checking does it basically render the 4.0.0 cncf.kubernetes provider, well, useless :)?

**Comment by potiuk** [comment]
(i.e. should we release 4.0.1 rather quickly ?)

**Comment by potiuk** [comment]
@dimon222 - what are the consequences of this ?

**Comment by potiuk** [comment]
@dimon222  - I'd love to merge that one, but would like to know the consequence ogf "period" - I have not analyzed it but is this a breaking change/fix ? What are the consequences?

**Comment by dimon222** [comment]
@potiuk 
The initial root cause why I care is described there:
https://github.com/apache/airflow/issues/16611
tl;dr hostname self-resolution problems since hostname of container looks like 2nd level DNS name.

In general, it shouldn't break much because returning back to hyphen is how it was before Airflow 2.0 upgrade that introduced it. Since the switch to hyphen happened in 95% of places of #19036 , I suspect we all agree that its intended to have hyphen. The last change required to make it happen is presented in this PR.

**Comment by potiuk** [comment]
Ok. Cool. It's a bugfix then :)
